### ISSUE TITLE
How to add different colors for startdate and enddate when enable range selection 

### ISSUE THREAD

**Issue by kobagapu** [stephy/CalendarPicker#261]
onDateChange(date, type) {
    if (type === 'END_DATE') {
      this.setState({
        selectedEndDate: date,
      });
    } else {
      this.setState({
        selectedStartDate: date,
        selectedEndDate: date,
      });
    }
  }
  
 <CalendarPicker
                      dayLabelsWrapper = {{borderTopWidth:0}}
                        startFromMonday={true}
                        allowRangeSelection={true}
                            minDate={minDate}
                            weekdays={['S', 'M', 'T', 'W', 'T', 'F', 'Sat']}
                        maxDate={maxDate}
                        //previousTitle="    <"
                        //nextTitle=">    "
                            previousTitleStyle={{ marginLeft: 30, fontSize: 30 }}
                            //customDatesStyles={customDatesStyles}
                        nextTitleStyle = {{fontSize: 30,marginRight:30}}
                        todayBackgroundColor={'#E9EFF4'}
                        selectedDayColor={'#E9EFF4'}
                        selectedDayTextColor="#242E42"
                            scaleFactor={450}
                            width={450}
                            allowBackwardRangeSelect = {true}
                        textStyle={{
                          fontFamily: Fonts.latoMedium,
                          color: '#242E42',
                        }}
                        onDateChange={this.onDateChange}
                          />

**Comment by kobagapu** [comment]
https://user-images.githubusercontent.com/61305724/103078765-8c1f4e80-45f8-11eb-846d-1e9119768dd8.mov

**Comment by kobagapu** [comment]
https://user-images.githubusercontent.com/61305724/103078934-eb7d5e80-45f8-11eb-949d-e4a24f3289ab.mov

**Comment by peacechen** [comment]
Try passing styles to these [props](https://github.com/stephy/CalendarPicker#calendarpicker-props):
> selectedRangeStartStyle | ViewStyle | Optional. Style for range selected start day.
> selectedRangeEndStyle | ViewStyle | Optional. Style for range selected end day.
> selectedRangeStyle | ViewStyle | Optional. Style for all days in range selection.

**Comment by kobagapu** [comment]
Thanks @peacechen  its working but is there any  prop?  changing text color for selectedRangeStartStyle and selectedRangeEndStyle

**Comment by kobagapu** [comment]
Thanks @peacechen its working  but is there any prop? changing text color in selectedRangeStartStyle and selectedRangeEndStyle 
![Screenshot 2020-12-28 at 11 00 07 AM](https://user-images.githubusercontent.com/61305724/103191734-0eab5500-48fc-11eb-872f-47cf882ef4b8.png)

**Comment by kobagapu** [comment]
> Try passing styles to these [props](https://github.com/stephy/CalendarPicker#calendarpicker-props):
> 
> > selectedRangeStartStyle | ViewStyle | Optional. Style for range selected start day.
> > selectedRangeEndStyle | ViewStyle | Optional. Style for range selected end day.
> > selectedRangeStyle | ViewStyle | Optional. Style for all days in range selection.

**Comment by peacechen** [comment]
Try the prop `selectedDayTextStyle`

**Comment by kobagapu** [comment]
> Try the prop `selectedDayTextStyle`

No its not working it does not change text color in selectedRangeStartStyle 

 selectedRangeStartStyle = {{ backgroundColor: Colors.themePrimary,selectedDayTextStyle:'#fff'}}

**Comment by peacechen** [comment]
That looks like a bug due to the order of styles in the array.
https://github.com/stephy/CalendarPicker/blob/b93ae2ee9b762f2fa6e446d0c86003d21197faf8/CalendarPicker/Day.js#L160

If you have time, please submit a PR that moves `propSelectedDayTextStyle` to the end. It should look like this on the 7 lines that use it:
```
          selectedDayTextStyle = [propSelectedDayTextStyle, styles.selectedDayLabel];
```

**Comment by peacechen** [comment]
Published in 7.0.7
Happy New Year 🎆

**Comment by kobagapu** [comment]
Thanks @peacechen  
<CalendarPicker
                      dayLabelsWrapper = {{borderTopWidth:0}}
                        startFromMonday={true}
                        allowRangeSelection={true}
                            minDate={minDate}
                            weekdays={['S', 'M', 'T', 'W', 'T', 'F', 'Sat']}
                        maxDate={maxDate}
                        previousTitle="    <"
                        nextTitle=">    "
                            previousTitleStyle={{ marginLeft: 30, fontSize: 30 }}
                            //customDatesStyles={customDatesStyles}
                        nextTitleStyle = {{fontSize: 30,marginRight:30}}
                        todayBackgroundColor={'#E9EFF4'}
                        selectedDayColor={'#E9EFF4'}
                        selectedDayTextColor="#242E42"
                            //scaleFactor={400}
                            width={450}
                            height = {350}
                            allowBackwardRangeSelect={true}
                            **selectedRangeStartStyle = {{ backgroundColor: Colors.themePrimary,selectedDayTextColor:'white'}}**
                            selectedRangeEndStyle={{ backgroundColor: Colors.themePrimary}}
                        textStyle={{
                          fontFamily: Fonts.latoMedium,
                          color: '#242E42',
                        }}
                        onDateChange={this.onDateChange}
                          />
its not working when add this prop   selectedDayTextColor:'white'

**Comment by peacechen** [comment]
Pass a style to `selectedDayTextStyle`

**Comment by kobagapu** [comment]
@peacechen selectedDayTextStyle pass it changes all selected days but i want to change selectedRangeStartStyle and selectedRangeEndStyle

**Comment by peacechen** [comment]
I see what you mean. `selectedRangeStartStyle` applies to the container View, not the Text within it, so it's unable to change the text color. New props `selectedRangeStartTextStyle` and `selectedRangeEndTextStyle` will need to be added. Please submit a PR that adds them.

**Comment by kobagapu** [comment]
https://github.com/stephy/CalendarPicker/issues/261#issuecomment-751582292

**Comment by peacechen** [comment]
The new props have been published in 7.0.8

I recommend that you learn how to submit Pull Requests (PR) to contribute to this library. You wouldn't need to wait around for someone else to make the change for you.

**Comment by kobagapu** [comment]
Thank you @peacechen
